onnx/utils.py

```diff
@@ -220,13 +220,11 @@ def extract_model(
     if check_model:
         onnx.checker.check_model(input_path)
 
-    if infer_shapes and os.path.getsize(input_path) > onnx.checker.MAXIMUM_PROTOBUF:
+    if infer_shapes:
         onnx.shape_inference.infer_shapes_path(input_path, output_path)
         model = onnx.load(output_path)
     else:
         model = onnx.load(input_path)
```

Comment:
You may load without external data first, infer shapes, then load external data for the model.

Reply:
you mean something like this?

Reply:
I'm thinking if we should do these things in the `infer_shapes_path` function or we can merge `infer_shapes` and `infer_shapes_path` https://github.com/onnx/onnx/pull/6479?

Reply:
@justinchuby which way do you prefer?

Reply:
Is there any way to know if a model has external data or not?

Reply:
One has to go through all tensors, which is a O(n) operation

Reply:
is it ok to call `onnx.load_external_data_for_model` if there's no external data?